modules/miner/update.go

```diff
@@ -28,11 +27,138 @@ func (m *Miner) addNewTxns(diff *modules.TransactionPoolDiff) {
 				totalFees = totalFees.Add(fee)
 			}
 		}
-		m.splitSets[m.setCounter] = &splitSet{
+		// We will check to see if this splitSet belongs in the block.
+		s := &splitSet{
 			size:         size,
 			averageFee:   totalFees.Div64(size),
 			transactions: newSet.Transactions,
 		}
+
+		elem := &mapElement{
+			set:   s,
+			id:    splitSetID(m.setCounter),
+			index: 0,
+		}
+
+		m.splitSets[splitSetID(m.setCounter)] = s
+		newElements = append(newElements, elem)
```

Comment:
The goal here is to make life easy for yourself as the programmer and for reviewers and other programmers. You want everything that's related all happening in one place, and then you split off the unrelated or fully-independent bits into separate functions.